hipe4ml/TrainingAndTesting/plot_utils.py

```diff
@@ -0,0 +1,310 @@
+def plot_output_train_test(
+        clf, x_train, y_train, x_test, y_test, model='xgb', features=None, bins=80):
+    '''
+    Plot the BDT output for the signal and bckground distributions
+    both for training and test set.
+
+    Input
+    ----------------------------------------
+    x_train: training set
+    y_train: training label
+    x_test: test set
+    y_test: test label
+    model: Could be 'xgb' or 'sklearn'
+    features: training columns
+
+    Output
+    ----------------------------------------
+    res: matplotlib object with the BDT output
+
+
+    '''
+
+    prediction = []
+    for xxx, yyy in ((x_train, y_train), (x_test, y_test)):
+        if model == 'xgb':
+            df1 = clf.predict(xxx[yyy > 0.5][features], output_margin=True)
+            df2 = clf.predict(xxx[yyy < 0.5][features], output_margin=True)
+        elif model == 'sklearn':
+            df1 = clf.decision_function(xxx[yyy > 0.5]).ravel()
+            df2 = clf.decision_function(xxx[yyy < 0.5]).ravel()
+        else:
+            print('Error: wrong model type used')
+            return None
+        prediction += [df1, df2]
+
+    low = min(np.min(d) for d in prediction)
+    high = max(np.max(d) for d in prediction)
+    low_high = (low, high)
+
+    res = plt.figure()
+
+    plt.hist(prediction[1], color='b', alpha=0.5, range=low_high, bins=bins,
+             histtype='stepfilled', density=True, log=True, label='Background pdf Training Set')
+    plt.hist(prediction[0], color='r', alpha=0.5, range=low_high, bins=bins,
+             histtype='stepfilled', density=True, log=True, label='Signal pdf Training Set')
+
+    hist, bins = np.histogram(
+        prediction[2], bins=bins, range=low_high, density=True)
+    scale = len(prediction[2]) / sum(hist)
+    err = np.sqrt(hist * scale) / scale
+    center = (bins[:-1] + bins[1:]) / 2
+    plt.errorbar(center, hist, yerr=err, fmt='o',
+                 c='r', label='Background pdf Test Set')
+
+    hist, bins = np.histogram(
+        prediction[3], bins=bins, range=low_high, density=True)
+    scale = len(prediction[2]) / sum(hist)
+    err = np.sqrt(hist * scale) / scale
+    plt.errorbar(center, hist, yerr=err, fmt='o',
+                 c='b', label='Signal pdf Test Set')
+
+    # plt.gcf().subplots_adjust(left=0.14)
+    plt.xlabel('BDT output', fontsize=13, ha='right', position=(1, 20))
+    plt.ylabel(r'                                Counts (arb. units)', fontsize=13)
+    plt.legend(frameon=False, fontsize=12)
+    return res
+
+
+def plot_distr(sig_df, bkg_df, column=None, figsize=None, bins=50, log=False):
+    """Build a DataFrame and create two dataset for signal and bkg
+
+    Draw histogram of the DataFrame's series comparing the distribution
+    in `data1` to `data2`.
+
+    Input
+    -----------------------------------------
+    sig_df: signal candidates dataframe
+    bkg_df: background candidates dataframe
+    column: training columns
+    log: log scale plot
```

Comment:
Not all the function inputs are described. Column is not related to the training columns.